# Rainfall Exploration

In [ ]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
import time
import seaborn as sns 
import rasterio
import os
from datetime import datetime, timedelta

In [7]:
kano_2020 = pd.read_csv("../data/kano_chirps_daily_timeseries_2020.csv")

In [ ]:
kano_2020.head()

system:index        date  rainfall                                    .geo
0      20200101  2020-01-01       0.0  {"type":"MultiPoint","coordinates":[]}
1      20200102  2020-01-02       0.0  {"type":"MultiPoint","coordinates":[]}
2      20200103  2020-01-03       0.0  {"type":"MultiPoint","coordinates":[]}
3      20200104  2020-01-04       0.0  {"type":"MultiPoint","coordinates":[]}
4      20200105  2020-01-05       0.0  {"type":"MultiPoint","coordinates":[]}

In [9]:
print(f"Min Rainfall Amount: {kano_2020['rainfall'].min()} mm")
print(f"Max Rainfall Amount: {kano_2020['rainfall'].max()} mm")


Min Rainfall Amount: 0.0 mm
Max Rainfall Amount: 38.23091941156892 mm


### Archived

In [ ]:
# with rasterio.open('../data/kano_chirps_export.tif') as src:
#     rainfall = src.read(1)
#     print(f"Shape: {rainfall.shape}")
#     print(f"CRS: {src.crs}")
#     print(f"Bounds: {src.bounds}")
#     print(f"Resolution: {src.res}")

#     print(f"Min rainfall: {np.nanmin(rainfall)} mm")
#     print(f"Max rainfall: {np.nanmax(rainfall)} mm")
#     print(f"Mean rainfall: {np.nanmean(rainfall)} mm")

Shape: (508, 590)
CRS: EPSG:4326
Bounds: BoundingBox(left=7.668019265244236, bottom=10.499059883146908, right=10.583052362212083, top=13.00895278697685)
Resolution: (0.0049407340626573685, 0.0049407340626573685)
Min rainfall: 1810.804957310429 mm
Max rainfall: 5740.217972217128 mm
Mean rainfall: 3563.244530492109 mm


Shape of (508, 590) means that we have a grid of 299,720 rainfall measurements arranged in 508 rows and 590 columns, covering the selected area (Kano in the above cell). Each number in the array is the rainfall amount (in mm)